test/penalizer/Utils.ts

```diff
@@ -0,0 +1,170 @@
+import { Commitment } from '../../src/enveloping/Commitment'
+import { AccountKeypair } from '../../src/relayclient/AccountManager'
+import { Address } from '../../src/relayclient/types/Aliases'
+import { RelayHubInstance, TestVerifierEverythingAcceptedInstance, TestTokenInstance, IForwarderInstance } from '../../types/truffle-contracts'
+import { getLocalEip712Signature } from '../../Utils'
+
+const gasPrice = '1'
+
+interface RelayRequestParams{
+  from: Address
+  to: Address
+  relayData: string
+  enableQos?: boolean
+}
+
+interface CommitmentReceipt {
+  workerAddress: string | BN
+  commitment: {
+    time: number | BN | string
+    from: string | BN
+    to: string | BN
+    relayHubAddress: string | BN
+    relayWorker: string | BN
+    enableQos: boolean
+    data: string
+    signature: string
+  }
+  workerSignature: string
+}
+
+export class RelayHelper {
+  relayHub: RelayHubInstance
+  relayOwner: Address
+  relayWorker: Address
+  relayManager: Address
+  forwarder: IForwarderInstance
+  verifier: TestVerifierEverythingAcceptedInstance
+  token: TestTokenInstance
+  chainId: number
+
+  constructor (
+    relayHub: RelayHubInstance,
+    relayOwner: Address,
+    relayWorker: Address,
+    relayManager: Address,
+    forwarder: IForwarderInstance,
+    verifier: TestVerifierEverythingAcceptedInstance,
+    token: TestTokenInstance,
+    chainId: number
+  ) {
+    this.relayHub = relayHub
+    this.relayOwner = relayOwner
+    this.relayWorker = relayWorker
+    this.relayManager = relayManager
+    this.forwarder = forwarder
+    this.verifier = verifier
+    this.token = token
+    this.chainId = chainId
+  }
+
+  async init (): Promise<void> {
+    await this.token.mint('1000', this.forwarder.address)
+
+    await this.relayHub.stakeForAddress(this.relayManager, 1000, {
+      from: this.relayOwner,
+      value: ether('1'),
+      gasPrice: gasPrice
+    })
+
+    await this.relayHub.addRelayWorkers([this.relayWorker], { from: this.relayManager, gasPrice: gasPrice })
+  }
+
+  async createRelayRequest (params: RelayRequestParams): Promise<RelayRequest> {
+    return {
+      request: {
+        relayHub: this.relayHub.address,
+        to: params.to,
+        data: params.relayData,
+        from: params.from,
+        nonce: (await this.forwarder.nonce()).toString(),
+        value: '0',
+        gas: '3000000',
+        tokenContract: this.token.address,
+        tokenAmount: '1',
+        tokenGas: '50000',
+        enableQos: params.enableQos ?? false
+      },
+      relayData: {
+        gasPrice: gasPrice,
+        relayWorker: this.relayWorker,
+        callForwarder: this.forwarder.address,
+        callVerifier: this.verifier.address,
+        domainSeparator: getDomainSeparatorHash(this.forwarder.address, this.chainId)
+      }
+    }
+  }
+
+  getRelayRequestSignature (relayRequest: RelayRequest, account: AccountKeypair): string {
```

Comment:
couldn't find an existing function for this, but this code (or something very similar) is all over the codebase